nltk/sentiment/sentiment_analyzer.py

```diff
@@ -47,10 +47,10 @@ def all_words(self, documents, labeled=None):
         all_words = []
         if labeled is None:
             labeled = documents and isinstance(documents[0], tuple)
-        if labeled == True:
-            for words, sentiment in documents:
-                all_words.extend(words)
-        elif labeled == False:
+        if labeled:
+            for words in documents:
+                all_words.extend(words[0])
```

Comment:
```suggestion
            for words, _sentiment in documents:
                all_words.extend(words)
```
I believe this is the best solution - this way no decent IDE or linter will complain about `_sentiment` not being used, but a developer can still read and understand that `document` must be a tuple of `(words, sentiment)`, which is valuable information in understanding the code.

Reply:
I believe that the below snippet is somehow misleading. 

```
for words, _ in documents:
    all_words.extend(words)
```
Since you removed the sentiment variable, so it does not clarify what it really does. 

I am okay with the changes, but I am going to leave it for a while, if someone has different POV.